File: backend/src/main/java/com/group6/sams/service/CourseService.java
package com.group6.sams.service;

import com.group6.sams.dto.request.CourseRequest;
import com.group6.sams.dto.response.CourseResponse;
import com.group6.sams.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface CourseService {

    PageResponse<CourseResponse> findAll(Long departmentId, Integer semester,
                                         Long teacherId, String search, Pageable pageable);

    CourseResponse findById(Long id);

    /** Courses assigned to the teacher profile linked to this user account. */
    List<CourseResponse> findMyCourses(Long userId);

    CourseResponse create(CourseRequest request);

    CourseResponse update(Long id, CourseRequest request);

    void delete(Long id);
}

File: frontend/src/services/courseService.js
import api from './api';

/** Course API. Owner: Member 2. */
const courseService = {
  list: (params = {}) => api.get('/courses', { params }).then((r) => r.data),
  get: (id) => api.get(`/courses/${id}`).then((r) => r.data),
  myCourses: () => api.get('/courses/my').then((r) => r.data),
  create: (payload) => api.post('/courses', payload).then((r) => r.data),
  update: (id, payload) => api.put(`/courses/${id}`, payload).then((r) => r.data),
  remove: (id) => api.delete(`/courses/${id}`),
};

export default courseService;
